types/node-ipc/index.d.ts

```diff
@@ -0,0 +1,297 @@
+         */
+        rawBuffer: boolean;
+        /**
+         * Default: false
+         * Synchronous requests. Clients will not send new requests until the server answers
+         */
+        sync: boolean;
+        /**
+         * Default: false
+         * Turn on/off logging default is false which means logging is on
+         */
+        silent: boolean;
+        /**
+         * Default: true
+         * Turn on/off util.inspect colors for ipc.log
+         */
+        logInColor: boolean;
+        /**
+         * Default: 5
+         * Set the depth for util.inspect during ipc.log
+         */
+        logDepth: number;
+        /**
+         * Default: 100
+         * This is the max number of connections allowed to a socket. It is currently only being set on Unix Sockets. Other Socket types are using the system defaults
+         */
+        maxConnections: number;
+        /**
+         * Default: 500
+         * This is the time in milliseconds a client will wait before trying to reconnect to a server if the connection is lost. This does not effect UDP sockets since they do not have a client server relationship like Unix Sockets and TCP Sockets
+         */
+        retry: number;
+        /*  */
+        /**
+         * Default: false
+         * if set, it represents the maximum number of retries after each disconnect before giving up and completely killing a specific connection
+         */
+        maxRetries: boolean;
+        /**
+         * Default: false
+         * Defaults to false meaning clients will continue to retry to connect to servers indefinitely at the retry interval. If set to any number the client will stop retrying when that number is exceeded after each disconnect. If set to true in real time it will immediately stop trying to connect regardless of maxRetries. If set to 0, the client will NOT try to reconnect
+         */
+        stopRetrying: boolean;
+        /**
+         * Primarily used when specifying which interface a client should connect through. see the socket.connect documentation in the node.js api https://nodejs.org/api/net.html#net_socket_connect_options_connectlistener
+         */
+        interfaces: {
+            /**
+             * Default: false
+             */
+            localAddress?: boolean;
+            /**
+             * Default: false
+             */
+            localPort?: boolean;
+            /**
+             * Default: false
+             */
+            family?: boolean;
+            /**
+             * Default: false
+             */
+            hints?: boolean;
+            /**
+             * Default: false
+             */
+            lookup?: boolean;
+        };
+        tls: {
+            rejectUnauthorized?: boolean;
+            public?: string;
+            private?: string;
+        };
+    }
+}
+interface RootIPC extends NodeIPC.IPC {
```

Comment:
If you export `RootIPC`, everything in `NodeIPC` becomes inaccessible from the outside. It will be visible with:
```ts
declare const NodeIPC: NodeIPC.IPC & { IPC: new () => NodeIPC.IPC };
export = NodeIPC;
```